agent/src/softwarecontaineragent.cpp

```diff
@@ -119,94 +120,113 @@ bool SoftwareContainerAgent::triggerPreload()
     return true;
 }
 
-inline bool SoftwareContainerAgent::isIdValid(ContainerID containerID)
+void SoftwareContainerAgent::assertContainerExists(ContainerID containerID)
 {
-    return (containerID < INT32_MAX)
-        && (containerID >= 0)
-        && (1 == m_containers.count(containerID));
+    if (containerID >= INT32_MAX) {
+        std::string errorMessage("Invalid Container ID: "
+                                + std::to_string(containerID)
+                                + ". ID can not be greater than INT32");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
+    }
+    
+    if (containerID < 0) {
+        std::string errorMessage("Invalid Container ID: "
+                                + std::to_string(containerID)
+                                + ". ID can not be negative");
+        log_error() << errorMessage;
+        throw SoftwareContainerError(errorMessage);
+    }
+    if (1 > m_containers.count(containerID)) {
```

Comment:
I do not like using `0` as `false` since they are fundamentally different types. Could go for `0 == m_containers.count(containerID)` but then negative numbers would be allowed. Although, negative numbers are not possible anyway but it still feels nice :)